client/src/components/VisibilityControl/VisibilityControl.tsx

```diff
@@ -9,7 +9,24 @@ import { secondaryColor, tertiaryColor } from "src/style_constants/index";
 
 import "./VisibilityControl.scss";
 
-export class VisibilityControl extends React.Component {
+interface VisibilityControlProps {
+  items: itemInterface[];
+  item_component_order: Record<string, unknown>;
+  click_callback: (key: string) => void;
+  show_eyes_override: boolean;
+}
+
+interface itemInterface {
+  active: boolean;
+  aria_text: string;
+  count: number;
+  key: string;
+  icon: HTMLElement;
+  text: HTMLElement;
```

Comment:
`icon` and `text` should be `React.ReactNode` (basically anything that is valid for the curly brackets inside of a JSX block to evaluate to; more JSX, a string, null, etc). 

Looks like the component also allows for them to be `undefined`, which the current typing doesn't accept (coincidentally, a `React.ReactNode` would) 

Maybe `text` should actually be a string, depends if the styles from `.visibility-control__text` make for a usage that should be constrained to text content.

Arguably count _could_ also be of type `React.ReactNode`, based on usage, although I think number is better. Just double check that that (and the rest of the typing) aligns with the current actual usage of this component\*.

\* important to check existing usage while typing, _but_ there's a balance to be struck between matching current usage and taking this as a chance to enforce tighter typing. Play it by ear.